sandbox/plc4c/examples/hello-subsciptions/src/hello_subscriptions.c

```diff
@@ -0,0 +1,264 @@
+          state = CONNECTED;
+        } else if (plc4c_connection_has_error(connection)) {
+          printf("FAILED\n");
+          return -1;
+        }
+        break;
+      }
+      case CONNECTED: {
+        // Create a new read-request.
+        printf("Preparing a read-request ... ");
+
+        result = plc4c_connection_create_subscription_request(
+            connection, &subscription_request);
+        if (result != OK) {
+          printf("FAILED\n");
+          return -1;
+        }
+        printf("SUCCESS\n");
+
+        printf("Adding cyclic item for 'RANDOM/foo:INTEGER' ... ");
+        result = plc4c_subscription_request_add_cyclic_item(
+            subscription_request, "RANDOM/foo:INTEGER", 500);
+        if (result != OK) {
+          printf("FAILED\n");
+          return -1;
+        }
+        printf("SUCCESS\n");
+
+        // Execute the read-request.
+        printf("Executing a read-request ... ");
+        result = plc4c_subscription_request_execute(
+            subscription_request, &subscription_request_execution);
+        if (result != OK) {
+          printf("FAILED\n");
+          return -1;
+        } else {
+          state = SUBSCRIPTION_REQUEST_SENT;
+        }
+        break;
+      }
+        // Wait until the read-request execution is finished.
+      case SUBSCRIPTION_REQUEST_SENT: {
+        if (plc4c_subscription_request_execution_check_finished_successfully(
+                subscription_request_execution)) {
+          printf("SUCCESS\n");
+          state = SUBSCRIPTION_RESPONSE_RECEIVED;
+        } else if (
+            plc4c_subscription_request_execution_check_finished_with_error(
+                subscription_request_execution)) {
+          printf("FAILED\n");
+          return -1;
+        }
+        break;
+      }
+      case SUBSCRIPTION_RESPONSE_RECEIVED: {
+        // Get the response for the given subscription-request.
+        plc4c_subscription_response *subscription_response =
+            plc4c_subscription_request_execution_get_response(
+                subscription_request_execution);
+        if (subscription_response == NULL) {
+          printf("FAILED (No Response)\n");
+          return -1;
+        }
+
+        // Iterate over all returned items.
+        plc4c_list_element *cur_element =
+            plc4c_utils_list_head(subscription_response->response_items);
+        while (cur_element != NULL) {
+          plc4c_response_subscription_item *subscription_item_item =
+              cur_element->value;
+
+          printf("Value %s (%s):", subscription_item_item->item->name,
+                 plc4c_response_code_to_message(
+                     subscription_item_item->response_code));
+
+          if (subscription_item_item->response_code == OK) {
+            // Save the subscription handles...
+            subscription_handle = subscription_item_item->subscription_handle;
+          }
+
+          cur_element = cur_element->next;
+        }
+
+        // Clean up.
```

Comment:
So this just reads one and then quits?  Wouldn't we want them to read until someone hit a key?

Reply:
The point of this example is not to build a good application, but to demonstrate the API ... so this counter demonstrates the api usage without any constraints about how to handle key presses.